pinot-server/src/main/java/org/apache/pinot/server/api/resources/InstanceResource.java

```diff
@@ -97,4 +103,22 @@ public Map<String, String> getInstancePools() {
     Map<String, String> pools = instanceConfig.getRecord().getMapField(InstanceUtils.POOL_KEY);
     return pools == null ? Collections.emptyMap() : pools;
   }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/diskUtilization")
```

Comment:
Does it makes sense to have a wrapper endpoint for all check that may come in future? Just single `resourceUtilization` endpoint that gives back the map of all resource stats. We can also have option to skip/includeOnly any resource computation by passing comma separated resource names in case we only want some targeted resource info (link in current case). Thoughts?

Reply:
Having a single endpoint was definitely considered. However, we would want to introduce the endpoint in the controller. The `diskUtilization` endpoint is a Server level API and not exposed in the controller itself. 

Currently, we have the `ResourceUtilizationManager` which checks disk utilization. Additional checks can be added to this class in the future. When `resourceUtilization` endpoint is introduced in the controller, the call would go via the `ResourceUtilizationManager` class and it could provide a single view of the various stats that are collected.